crucible/src/Lang/Crucible/Types.hs

```diff
@@ -455,8 +457,47 @@ instance HashableF TypeRepr where
 instance Hashable (TypeRepr ty) where
   hashWithSalt = $(U.structuralHashWithSalt [t|TypeRepr|] [])
 
+-- | Pretty-print a type.
+--
+-- This instance attempts to be consistent with the syntax provided in the
+-- @crucible-syntax@ package.
 instance Pretty (TypeRepr tp) where
-  pretty = viaShow
+  pretty x =
+    let prettyCtx :: Ctx.Assignment TypeRepr ctx -> Doc ann
+        prettyCtx = hsep . toListFC pretty in
+    let prettyBaseCtx :: Ctx.Assignment BaseTypeRepr ctx -> Doc ann
+        prettyBaseCtx = hsep . toListFC pretty in
+    case x of
+      AnyRepr -> "Any"
+      UnitRepr -> "Unit"
+      BoolRepr -> "Bool"
+      NatRepr -> "Nat"
+      IntegerRepr -> "Integer"
+      RealValRepr -> "RealVal"
+      ComplexRealRepr -> "ComplexReal"
+      BVRepr n -> parens ("Bitvector" <+> viaShow n)
+      IntrinsicRepr name tys ->
+        parens (pretty (symbolRepr name) <+> prettyCtx tys)
+      RecursiveRepr name tys ->
+        parens (pretty (symbolRepr name) <+> prettyCtx tys)
```

Comment:
We use the same pretty-printing logic to print `IntrinsicRepr`s and `RecursiveRepr`s, but I worry that it will be easy to confuse the two. Is it worth thinking of a convention to distinguish the two?

Reply:
Yeah! Maybe we prefix recursive types with `Rec`? So you'd get `(Rec <name> args)`?

Reply:
I like the `(Rec <name> args)` suggestion.